pypeit/spectrographs/gemini_gmos.py

```diff
@@ -268,75 +347,171 @@ def get_rawimage(self, raw_file, det):
-        for kk, jj in enumerate(order):
-            # grab complete extension...
-            data, overscan, datasec, biassec, x1, x2 = gemini_read_amp(hdu, jj)
-            # insert components into output array...
-            inx = data.shape[0]
-            xs = inx * kk
-            xe = xs + inx
-
-            # insert data...
-            # Data section
-            #section = '[{:d}:{:d},:]'.format(xs * xbin, xe * xbin)  # Eliminate lines
-            #dsec.append(section)
-            array[xs:xe, :] = np.flipud(data)
-            rawdatasec_img[xs:xe, :] = kk+1
-
-            # ; insert postdata...
-            xs = nx - numamp * nxb + kk * nxb
-            xe = xs + nxb
-
-            #osection = '[{:d}:{:d},:]'.format(xs * xbin, xe * xbin)  # TRANSPOSED FOR WHAT COMES
-            #osec.append(osection)
-            array[xs:xe, :] = overscan
-            oscansec_img[xs:xe, :] = kk+1
+#        # TODO: Why is this stuff here and not in the relevant subclass?
+#        if numamp == 2:  # E2V
+#            if det == 1:  # BLUEST DETECTOR
+#                order = range(6, 4, -1)
+#            elif det == 2:  # NEXT
+#                order = range(3, 5)
+#            elif det == 3:  # REDDEST DETECTOR
+#                order = range(1, 3)
+#        elif numamp == 4:  # Hamamatsu
+#            if det == 1:  # BLUEST DETECTOR
+#                order = range(12, 8, -1)
+#            elif det == 2:  # BLUEST DETECTOR
+#                order = range(8, 4, -1)
+#            elif det == 3:  # BLUEST DETECTOR
+#                order = range(4, 0, -1)
+#        else:
+#            msgs.error(f'Unexpected number of amplifiers {numamp}; expected 2 or 4!')
+
+        # Get the HDU read order for this instrument
+        order = self.hdu_read_order()
+        for ii in range(nimg):
+
+            # insert extensions into master image...
+            for kk, jj in enumerate(order[_det[ii]-1]):
+                # grab complete extension...
+                data, overscan, datasec, biassec, x1, x2 = gemini_read_amp(hdu, jj)
+                # insert components into output array...
+                inx = data.shape[0]
+                xs = inx * kk
+                xe = xs + inx
+
+                # insert data...
+                # Data section
+                #section = '[{:d}:{:d},:]'.format(xs * xbin, xe * xbin)  # Eliminate lines
+                #dsec.append(section)
+                array[ii,xs:xe,:] = np.flipud(data)
+                rawdatasec_img[ii,xs:xe, :] = kk+1
+
+                # ; insert postdata...
+                xs = nx - numamp * nxb + kk * nxb
+                xe = xs + nxb
+
+                #osection = '[{:d}:{:d},:]'.format(xs * xbin, xe * xbin)  # TRANSPOSED FOR WHAT COMES
+                #osec.append(osection)
+                array[ii,xs:xe,:] = overscan
+                oscansec_img[ii,xs:xe,:] = kk+1
 
         # Need the exposure time
+        # TODO: Why are we not using get_meta_value?
```

Comment:
Probably just for simplicity.  But probably best to use `get_meta_value()`

Reply:
Changed.